fix: Table | Inline Editing | Rehaul

spark-c opened the pull request:
Rehaul of the Table widget's [inline editing](https://docs.appsmith.com/reference/widgets/table/inline-editing) page. Structural, content, media, grammar/linter fixes.

The most significant changes were made to the **Update mode / Single row / Multi row** sections, so these will probably need some closer looks than the rest of the doc.

Resolves appsmithorg/appsmith#18170

jnikhila commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
-For example, add a regular expression for entering a name. The name can contain only alphabets and spaces between the first and last name.
+When you add a regular expression to a column, all user input in that column is compared to the pattern; it's considered to be valid when it matches the expression, or invalid when it doesn't match. When input is invalid, the cell displays its **Error message** to the user in a tool-tip.
+
+For example, add a regular expression for entering a name. The name can contain only alphabetical characters and spaces between the first and last name:
```
Adding a screenshot might help here. Any one screenshot showing an error message in a tooltip should be good.

jnikhila commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
 
-If a word other than "John" is added to the cell, an error will be displayed. Similarly, the above-mentioned binding variables can be used to obtain values and row indexes.&#x20;
-
+If a value other than "John" is added to the cell, an error is displayed. Similarly, the previously mentioned binding variables can be used to obtain other values and row indexes.&#x20;
```
Could we add a cross-reference to the binding variables section in the doc? It would help users to look at them if they need to.

On second thought, maybe you could add a cross-reference to the error message section below. It already has a screenshot there.

jnikhila commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
-The error message displays if the regular expression(regex) or valid property check fails. If a user enters an incorrect value, the widget shows a message "invalid input." You can change this message by using the `Error message` property to provide better feedback on the input given by the user.
+The error message appears if the regular expression (**Regex**) and/or the **Valid** property determine the input is invalid. If a user enters an incorrect value, the widget shows "invalid input." by default. You can change this message by using the **Error message** property to provide better feedback to the user.
 
 ![](/img/inline32.PNG)
```
Could we rename the image(inline32.PNG) to a meaningful name?

jnikhila commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -210,43 +186,195 @@ This property contains all the details of the edited rows. It has the following
-### updatedRowIndices
+#### updatedRowIndices
 
 This binding property displays the index number of the updated row. It contains an array of edited row indices.&#x20;
```
Probably some garbage value(&#x20;)

jnikhila reviewed (changes requested):
Hey @spark-c - I added few comments. Do let me know if you would like to discuss?

spark-c commented on website/docs/reference/widgets/table/README.md:
```diff
@@ -29,6 +29,7 @@ These properties allow you to edit the table itself. All these properties are pr
 | **Default Row Height**         | Sets the height of the row in the table - Default, short or tall.                                                                                                                                       |
 | **Enable multi-row selection** | Allows multiple rows of a table to be selected. The rows are populated in the selectedRows field.                                                                                                       |
 | **Table Record Count**         | The user has to input the total number of records returned as part of a query or API call, which will be displayed in a table. It is only visible when you choose to enable **server-side pagination**. |
+| **Allow adding a row**         | Toggles a button in the table which allows users to submit new rows of data. Only columns marked as **Editable** can accept user input. Use the **onSave** event to update the source of the table's data and reflect the user's changes. |
```
> `the **onSave** event updates the table data source (binds to the table data property).`

I think this might be a little misleading, b/c the event doesn't actually do anything on it's own. Maybe updating the original to something like:

`Use code or a query in the **onSave** event to update the source of the table's data and reflect the user's changes.`

The biggest idea I want to make sure we communicate to user is that the "adding rows" feature with tables does not actually make any changes to the table or data. The user must actually write code/queries in the **onSave** event in order to make anything happen.

spark-c replied:
(I made the update above, hopefully this will help make it clearer, or we can revisit later)

jnikhila replied:
@spark-c - as you pointed out (below), could you please add this information?

> The biggest idea I want to make sure we communicate to user is that the "adding rows" feature with tables does not actually make any changes to the table or data. The user must actually write code/queries in the onSave event in order to make anything happen.

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
-## How to edit a cell
-
-An edit icon appears on hovering an editable cell, which, when clicked, would make the cell editable.&#x20;
+Inline edit-ability for a column [can be enabled](#editing-cells) for the whole column or at the cell level based on a condition. You may also allow users to [update existing rows](#row-level-mode) of data, or [add new rows](#adding-new-rows) entirely.
```
Updated to something similar, so that it matches the wording in table/README

danciaclara reviewed (commented):
I'll review this post the Add new rows in table part is out on the docs

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
+| **updatedRow**               | Binding     | Contains the data of the row that triggered the **onSubmit**, **onSave** or **onDiscard** action. | `{{Table1.updatedRow}}`|
 
-![](</img/Update_mode.png>)
+#### Update mode (single row)
```
@jnikhila 

I like this idea (it's what I attempted originally), however I ran into the issue of the size/appearance of the headings on the page. In the above strategy, "Single-row" and "Multi-row" would both be 5th level headings (#####), which appear _smaller on the page_ than even regular bold text.

For example, 4th level headings are already the same size as bold text:

![as_bold](https://user-images.githubusercontent.com/51402465/203237556-8451cc91-604b-44cd-9919-6b1ef6eb8163.png)

I tried just using the bold text alone to separate the subheadings, but since each of those mode sections include guides, steps, pictures, etc -- It was very hard to parse and find the separation between them. (Which is why there's still bold text _and_ subheadings, to help make it as easy as possible to find the border btwn them).

Let me know what you think, maybe there's a better solution.

jnikhila replied:
For such headings we can add our own styles to change the default behaviour of Docusaurus. Do you think, you can give it a go and see if you could do anything here?

spark-c replied:
Sure! I added css rules for `h1` thru `h5`, setting `font-size` values to steps between `3rem` and `1.25rem`.

Net effect:
- `h1` page titles remain the same size, and all other subheadings `h2` - `h5` are a little bit larger.
- `h5` headings are larger than regular bold text
- `h4` headings (almost all of our detailed properties descriptions below properties tables are `h4`) are easier to pick out from surrounding content, which I think is nice.

jnikhila replied:
Awesome. @danciaclara - Do you want to take a look and update?

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
+#### Update mode (single row)
 
-### Row level mode
+**Single row** - Cells can be edited within a single row at a time, and saved using the Save/Discard column buttons that have been configured to run a query to submit a single row to a datasource. While a row has pending edits, the user isn't allowed to begin editing other rows.
```
Updated: 

**Single row** - Cells can be edited within a single row at a time, and then saved using the Save/Discard column buttons. While a row has pending edits, the user isn't allowed to begin editing other rows. Be sure to configure the Save/Discard buttons' **onSave** event to run a query that updates the table's datasource.

---

I also placed this in a quote block to more clearly separate it from the following content without being too visually distracting like a `note` or `info` block. (I did the same in the multi-row section). These first little bits are meant to be self-contained, brief-but-descriptive outlines of how the mode works, and then the following content goes into more detail.

danciaclara commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -349,10 +384,9 @@ To update the original datasource with the new data, you might execute a query l
-* [Appsmith Framework](../../appsmith-framework/)
-* [JavaScript Editor](../../../core-concepts/writing-code/javascript-editor-beta/)
+* [Sample apps for Tables](/learning-and-resources/sample-apps#table)
+  * [Editable table rows](https://app.appsmith.com/app/editable-table/row-save-62d8f8d0e1c2ed505a0557cc)
```
As there is not a lot of stuff we are giving them to refer, we can remove this section and have the sample app (Editing table rows) linked below the properties and events table. @pranavkanade can we also add adding new rows to the sample app, please?

danciaclara replied:
If there are troubleshooting issues/errors related to inline editing, they should come here

danciaclara commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -213,36 +200,84 @@ This property contains all the details of the edited rows. It has the following
+To save the edits made to your table rows, you'll need to set the table's [**Update mode**](#properties) to [**Single row**](#update-mode-single-row) or [**Multi row**](#update-mode-multi-row) mode and then configure your Save buttons to execute queries.
+:::
+
+### Perform an action when a cell is edited
```
sub-heading not required. it is continuous flow of steps that the user has to perform ot update the row after cell contents are edited

danciaclara commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -213,36 +200,84 @@ This property contains all the details of the edited rows. It has the following
+
+![](</img/Edit_icon_on_column_header.png>)
+
+### How to edit a cell
```
Let's discuss this

danciaclara reviewed (changes requested):
Just to ensure that the comments are not confusing, adding the proposed structure below

Inline Editing
   - Intro with video
   - Properties
   - Events
   - Editing cells
      - Intro
      -  not a heading [below intro explain Properties in detail under relevant sub-section]
           - How to edit a cell (everything under Enable editing cells should also come here as it is about the **Editable** 
              property)
           - Cell Wrapping
           - Validation properties (note Min, Max is only available for certain data types. check this)
   - Updating rows
      - Intro
      -  not a heading [below intro explain Properties in detail under relevant sub-section]
          - How to update a row (keep the explanation for the properties here for updatedRows, updatedRowIndices, 
             updatedRow, onSubmit event)

   - Adding rows (as is)

danciaclara commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -213,36 +200,84 @@ This property contains all the details of the edited rows. It has the following
-| **onSave**    | Action that gets triggered when the user clicks the save button for a cell or a new row.                                                       |
-| **onDiscard** | Action that gets triggered when user clicks discard button for a cell or a new row.                                                       |
+| **onSubmit**  | Action that gets triggered when the user moves away from editing a cell by pressing the enter key or clicking outside of the cell. |
+| **onSave**    | Action that gets triggered when the user clicks the save button for a cell or a new row.                               |
```
note that onSave and onDiscard is now only available for Adding new rows and no longer available for cell editing. it's only onSubmit.

spark-c replied:
@danciaclara Are they going to remove the **Save/Discard** button column in the table while in **Single row** edit mode?

From what I can see on notion it looks like they plan to keep the onSave stuff -- I think it was my description that was confusing. I updated it so it doesn't mention cells.

jnikhila commented:
@danciaclara - only the Title in Titlecase, along with product names and other exceptions that we could discover on the way. For all other headings we continue to use sentence case.

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,117 +1,105 @@
+# Inline editing
 
-Inline editing for a column can be enabled for the whole column or at the cell level based on a condition. You may also allow users to [update existing rows](#row-level-mode) of data, or [add new rows](#adding-new-rows) entirely.
+Inline editing for a column can be enabled for the whole column, or you can use code to enable it for only particular cells. You may also allow users to [update existing rows](#row-level-mode) of data, or [add new rows](#adding-new-rows) entirely.
```
While there isn't necessarily a switch for each specific cell, you can go to a column's settings and use code in the **Editable** property to specify which particular cells of that column will be editable.

For example,
- you might only want cells containing a particular value to be editable:
   ```javascript
   {{ currentRow.status === "pending" }}
   ```
- or maybe you might want a cell to be editable only if the sum of a couple of other cells meet a threshold:
   ```javascript
   {{ currentRow.account1 + currentRow.account2 > 1500 }}
   ```
- Or (for some reason) you could even only let even rows of the table be editable:
   ```javascript
   {{currentIndex % 2 === 0? true:false }}
   ```

danciaclara replied:
Yes and we should explain this in detail like you just did in your comment above in the Editing cells section, as your explanation was crystal clear to me :)  So please do include it in that section 

It just seemed a bit confusing in the intro. We want to keep it as simple as possible there so users get a clear idea of the feature or concept

danciaclara replied:
Hey I see the introductory text still the same as before. Maybe I'm looking at the wrong preview

spark-c replied:
Updated to match suggestion :)

spark-c commented on website/docs/reference/widgets/table/README.md:
```diff
@@ -68,9 +69,9 @@ This property shows all the columns in the table, and it gets auto-populated fro
 The data fetched from the Query/ API is sometimes too large to be displayed on one table page. Server-Side Pagination lets you implement pagination by limiting the number of results fetched per API / Query request. For more information on how to paginate your data, click [here](./../../../core-concepts/data-access-and-binding/displaying-data-read/display-data-tables.md#pagination).
 
-#### Inline Editing
+#### Inline editing
```
I thought we had settled on Title case for h1/(#) page titles, and then sentence case for all other headings? I may have missed an update.

danciaclara replied:
this is the main title right?

danciaclara commented:
> @danciaclara - only the Title in Titlecase, along with product names and other exceptions that we could discover on the way. For all other headings we continue to use sentence case.

@jnikhila So the one shown in the screenshot below should in title case right?
![Uploading Screenshot 2022-11-26 at 09.32.23.png…]()

jnikhila commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,197 +1,173 @@
 
-Based on the Column type, cell content can be edited. Once done user can move away from the edit mode in two ways.
+## Editing cells
 
```
I think that would be good.

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -213,36 +200,84 @@ This property contains all the details of the edited rows. It has the following
+
+You can also use JS to control the `Editable` property. For example, you might use this feature to make only a subset of the cells editable:
+
+![](</img/Using_JS_in_Editable.png>)
```
@danciaclara I have no idea why it seems to do this, but it goes back-and-forth on how the image renders. I've definitely seen the images super big / taking up the whole width of the page -- but currently they are just staying their true small size and behaving well. I'm not sure what makes them swap like they do.

![as_img_size](https://user-images.githubusercontent.com/51402465/204756243-028fe929-d4b9-4282-b123-2375a3b07df4.png)

danciaclara replied:
@spark-c did this get resolved when you specified width and height in the markdown?

spark-c replied:
I believe the issue will be resolved with this; I can't test it for sure on my end because the issue was only showing in the deploy preview. However, I tested it on the deploy preview with the browser style inspector and it worked in that way.

I'll push these changes and we'll see what the new preview looks like! Will update

spark-c replied:
It looks like the images are behaving now!

danciaclara commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,117 +1,105 @@
 
-1. Edited contents can be persisted on the Table cell by either pressing enter key or clicking anywhere outside the cell.
-2. You can discard the Edited contents by pressing the escape key.
+| Property                                           | Type        | Definition                                                                                                                                                                                                                                                               | Code Snippet                   |
```
Marking this as unresolved as some properties are jumbled. for eg: 'Update mode' should come after all the properties related to editing cells and just above updatedRows. Similarly, Allow adding a row should come above 'isAddRowInProgress' .  while maintaining order, also ensure they are grouped together too logically

spark-c replied:
Fixed

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
-![Editable Checkbox](</img/Editable_checkbox_-_Inline_editing.png>)
+The column list on the Table property pane has a checkbox to make that column editable.
 
 Click the checkbox inside a column card makes all the cells in that column editable. The editable checkbox at the top of the list can be checked to make all the editable columns editable.
```
updated, removed the first of those sentences and fixed grammar for the other

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
 ![](</img/Editable_preoperty.png>)
 
-JS option of the `Editable` property can be leveraged to make a subset of the cells in a column editable.
+Alternatively, A column can be editable by turning on `Editable` property inside the column's settings.
```
updated

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
+| **onSave**    | Triggered when the user clicks the save button for a new or existing row.                                                |
+| **onDiscard** | Triggered when the user clicks the discard button for a new or existing row.                                             |
+
+To try this feature yourself, take a look at the sample app for [editable table rows](https://app.appsmith.com/app/editable-table/row-save-62d8f8d0e1c2ed505a0557cc).
```
Moved to below video

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
 ![](</img/Using_JS_in_Editable.png>)
 
-Once a column has been made editable, an edit icon appears on the column header as an indicator.
+- You might only want cells containing a particular value to be editable:
```
Updated

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
 
-1. `onSubmit` property of the editable column can be used to save that cell when it's edited.
-2. Users can trigger save when a button is clicked outside the table.
+#### Valid
```
Grouped under heading name "**Input validation**" for clarity, and to help avoid confusion with the `Valid` property itself.

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
 
 :::info
-Validation ensures that only certain values are entered in a field. You have to make the column editable to allow validation. The validation feature is available for **text** and **numeric** column types.
+**Min** and **Max** properties are available for **numeric**  column types.
```
I tried something slightly different to see how it looked/felt -- I added a sentence to both properties that simply says:

> Only available for columns that are type **numeric**.

I did this because it felt odd to have at the bottom of the second property, when the user might never scroll that far if they only look at the first one. And if they only look at **Min**, they should still be able to find out that it's only available for **numeric** columns.

Anyway, here is the new **Min** and **Max** bits for example:

> ### Min
> Sets the minimum allowed value. For example, you could set the minimum value to 2 if you only want values greater than 2. Any number entered that's less than 2 is considered invalid. Only available for columns that are type **numeric**.

> ### Max
> Sets the maximum allowed value. For example, you could set the maximum value to 100 if you only want values less than 100. Any number entered that's more than 100 is considered invalid. Only available for columns that are type numeric.

Let me know how that feels!

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
-1. `currentRow` - gives access to the row values of the editable cell.
-2. `currentIndex` - index of the current editable row.
-3. `editedValue` - the newly entered value on the editable input.
+> Cells can be edited within a single row at a time, and then saved using the Save/Discard column buttons. While a row has pending edits, the user isn't allowed to begin editing other rows. Be sure to configure the Save/Discard buttons' **onSave** event to run a query that updates the table's datasource.
```
Fixed

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
+Read further to see how to [save the changes](#saving-changes) you've made.
 
-Actions can be performed at the row level. On selecting row level, a new column is injected called `Save/Discard` with two buttons when a column is made editable.
+### Related properties
```
Updated

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
 :::
 
-#### Custom mode
+#### Cell wrapping
```
Added

spark-c commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,198 +1,260 @@
+---
 
-If you enter a value other than an alphabet or space (number of special characters), the widget shows an error message "invalid input."
+## Update modes
```
I think I've made the necessary fixes/updates to match this structure; or at least it should be closer now :) We can take a look and get on a call again if there is more to be done.

danciaclara reviewed (changes requested):
@spark-c Could you redo the videos without the recording icon showing? 
<img width="884" alt="Screenshot 2022-12-07 at 11 09 32" src="https://user-images.githubusercontent.com/32227135/206097639-d96d9651-5f08-44f2-abf7-d866534f678c.png">

danciaclara commented:
@dilippitchika Could you do a quick tech review of this page to ensure we have not missed or miscommunicated any concept? Thanks 🙏 

Here's the preview - https://appsmith-docs-git-fix-table-inline-rehaul-get-appsmith.vercel.app/

dilippitchika commented:
Few comments
1. Cell wrapping property - It shouldn't be a part of inline editing but the table itself
2. We should think if we should the page call it "adding and updating data in tables" or something, because now it's not just inline editing anymore but both creating and editing

Adding more comments directly inline

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,193 +1,236 @@
+| -------------------------------------------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------ |
+| [**Editable**](#editable)                 | Formatting  | Controls whether cells of the column are editable                                                             | NA                             |
+| [**Cell Wrapping**](#cell-wrapping)            | Formatting  | Controls how overflowing contents of the column are handled.<br/> **on** - Contents get wrapped to the next line.<br/> **off** - Contents are truncated with an ellipsis. | NA |
+| [**Min**](#min)                      | Validation  | Sets the minimum allowed value.                                                                               | NA                             |
```
Can we start adding the column type this field will be visible for? Like min max will only be visible for number columns

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,193 +1,236 @@
 ## Editing cells
 
-The column list on the Table property pane has a checkbox to make that column editable.
+This feature enables users to make changes directly in the table cell. Currently, only a few column types support inline editing - **Text**, **Number**, **Switch**, and **Checkbox**. 
```
We also have select column type

dilippitchika replied:
Also please add a note that custom columns don't support editing as of now

danciaclara replied:
@spark-c I think you added this in the other place where this text occurs. Please make the changes here. and I think we can remove the second time it is repeated. 
<img width="953" alt="Screenshot 2022-12-12 at 18 17 03" src="https://user-images.githubusercontent.com/32227135/207048895-310e6f65-f21f-4a72-a2ff-59f635b65afb.png">

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,193 +1,236 @@
+These edits are only reflected on the Table widget UI and aren't updated in the source of truth database. Learn how to [save the changes](#saving-edits) you've made.
 
-Based on the Column type, cell content can be edited. Once done user can move away from the edit mode in two ways.
+#### Cell wrapping
```
Please move this to Table doc, this is not relevant for inline editing

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,193 +1,236 @@
-You can edit Multiple rows in custom update mode.
-:::
+<!-- ![](/img/table_err_msg.PNG) -->
+<img src="/img/table_err_msg.png" alt="Error message when input is invalid" width="524"/>
```
It says image not found, please check

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -208,29 +251,30 @@ This property contains all the details of the edited rows. It has the following
 #### updatedRow
 
-This property consists of the details of the row that recently got updated. Irrespective of the update mode, the `updatedRow` property consists of the row that got recently updated.
+This property contains the details of the row that was recently updated, regardless of the **Update mode**. For example, if you bind this property into a text widget, you get an output something like this:
```
Can we add that this is useful for single row update mode?

danciaclara replied:
and @spark-c maybe we should move it where we are explaining Single Row Mode. That way we are also visually segregating them.

<img width="1064" alt="Screenshot 2022-12-12 at 18 13 38" src="https://user-images.githubusercontent.com/32227135/207048250-6fb2be69-ef1a-43dd-9c78-85152d8ff042.png">

danciaclara replied:
@jnikhila is it ok update the CSS of the headings?

jnikhila replied:
Yes, we discussed it in one of our standup calls for this. @spark-c - As discussed earlier, have we taken into account the font-size ticket #590  that's already in the backlog, and will the CSS changes resolve that too?

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,193 +1,236 @@
 #### updatedRows
 
-This property contains all the details of the edited rows. It has the following structure.
+This property contains all the details of the edited rows. It has the following structure:
```
Can we add that this will only be useful for multi-row updates?

dilippitchika reviewed (commented):
Can we also update the sample data shown in tables to be mock data and not the default data on tables? @danciaclara

spark-c commented:
From @dilippitchika above:
> We should think if we should the page call it "adding and updating data in tables" or something, because now it's not just inline editing anymore but both creating and editing

I think the current title **Inline Editing** (while not necessarily _perfect_) might be okay. Even though there is now functionality to add/create rows -- we are still making edits to an existing dataset from within the widget itself. I think it still covers the features well enough, and we don't have to worry about any confusion that might come with renaming a page that's named after a prominent feature.

I'm open either way though, @danciaclara @jnikhila do you have thoughts?

dilippitchika commented:
I am worried it will be a discoverability problem, but as long as that doesn't happen i am fine either way

jnikhila commented:
> 
@spark-c - I agree with @dilippitchika  that the content is not just inline editing. IMO from the user’s perspective, it’s good to change the title (`Adding and Updating Data` [Table keyword is not needed as it's a subpage under the Table widget]).

jnikhila commented:
> @spark-c - I agree with @dilippitchika that the content is not just inline editing. IMO from the user’s perspective, it’s good to change the title (`Adding and Updating Data` [Table keyword is not needed as it's a subpage under the Table widget]).

We just have to ensure that the slug remains the same so that we don't have to add a redirect URL.

danciaclara commented:
> > 
> 
> @spark-c - I agree with @dilippitchika that the content is not just inline editing. IMO from the user’s perspective, it’s good to change the title (`Adding and Updating Data` [Table keyword is not needed as it's a subpage under the Table widget]).

I feel Inline editing is a very commonly used term. A search on Google will confirm this.  Do we need to change this?

spark-c commented:
> 1.    Remove the cell wrapping property from this page and move it to the main table page, i still see it here
> 2.   updatedRows property - Inform the user that updatedRows are to be used in multi column edit mode
> 3.   updatedRow property - inform the user that updatedRow is to be used in single column edit mode
> 4.    The callback to refresh the table which is mentioned in multi row update should also be mentioned in the single row update mode and add new row once onSave is clicked
> 5.    I am not sure if onSubmit will get triggered if the user is adding something in the new row. Can you check once?

@dilippitchika Sorry about those errors!

1. Some of the content was moved, but I forgot to remove it from the original table -- Fixed
2. Added another mention to the properties table
3. Added another mention to the properties table
4. Added instruction for including a callback to those sections
5. Nice catch! I tested it and you're totally right. Removed the part about **onSubmit** from Adding New Rows.

Thank you!

dilippitchika commented on website/docs/reference/widgets/table/inline-editing.md:
```diff
@@ -1,279 +1,316 @@
+  )
+}}
+
+###### updatedRow
```
@spark-c this is not being rendered accurately in the dp. Can you check once?

spark-c replied:
Oh heck, great catch. I was missing the backticks to terminate the code block -- Fixed
